src/pages/resultsView/plots/PlotsTabUtils.tsx

```diff
@@ -957,65 +960,81 @@ function makeAxisDataPromise_Clinical(
 
 function makeAxisDataPromise_Molecular(
     entrezGeneId: number,
-    molecularProfileId: string,
+    molecularProfileIdSuffix: string,
+    dataType: string,
     mutationCache: MobxPromiseCache<{ entrezGeneId: number }, Mutation[]>,
     numericGeneMolecularDataCache: MobxPromiseCache<
         { entrezGeneId: number; molecularProfileId: string },
         NumericGeneMolecularData[]
     >,
     entrezGeneIdToGene: MobxPromise<{ [entrezGeneId: number]: Gene }>,
-    molecularProfileIdToMolecularProfile: MobxPromise<{
-        [molecularProfileId: string]: MolecularProfile;
-    }>,
     mutationCountBy: MutationCountBy,
     coverageInformation: MobxPromise<CoverageInformation>,
-    samples: MobxPromise<Sample[]>
+    samples: MobxPromise<Sample[]>,
+    molecularProfileIdSuffixToMolecularProfiles: MobxPromise<{
+        [molecularProfileIdSuffix: string]: MolecularProfile[];
+    }>
 ): MobxPromise<IAxisData> {
-    let promise: MobxPromise<any>; /* = ;*/
+    let promises: MobxPromise<any>[] = []; /* = ;*/
     return remoteData({
         await: () => {
             const ret: MobxPromise<any>[] = [];
-            if (molecularProfileIdToMolecularProfile.isComplete) {
-                if (
-                    molecularProfileIdToMolecularProfile.result![
-                        molecularProfileId
-                    ].molecularAlterationType ===
-                    AlterationTypeConstants.MUTATION_EXTENDED
-                ) {
-                    // mutation profile
-                    promise = mutationCache.get({ entrezGeneId });
-                    ret.push(coverageInformation);
-                    ret.push(samples);
-                } else {
-                    // non-mutation profile
-                    promise = numericGeneMolecularDataCache.get({
-                        entrezGeneId,
-                        molecularProfileId,
-                    });
+            if (molecularProfileIdSuffixToMolecularProfiles.isComplete) {
+                const profileIds = molecularProfileIdSuffixToMolecularProfiles.result![
+                    molecularProfileIdSuffix
+                ].map(profile => profile.molecularProfileId);
+
+                // only push promise to promises when promises are empty
```

Comment:
@adamabeshouse @dippindots this seems kind of strange to me (though i see it's been around for a while).  we rely on state of promises variable which is local to the function, but that function can presumably be called multiple times.  what controls whether the function gets called redundantly?

Reply:
you're right, this was broken, because await and invoke are not necessarily in sync. we need to recalculate `promises` in `invoke`